R/zchunk_LB125.LC_tot.R

```diff
@@ -65,38 +64,132 @@ module_aglu_LB125.LC_tot_DISABLED <- function(command, ...) {
     #
     # ===================================================
 
+    # -----------------------------------------------------------------------------
+    # 2. Perform computations
+    #Create a table with all land types
+    #NOTE: If protected lands are included, uncomment the text below
+    bind_rows(L122.LC_bm2_R_HarvCropLand_Yh_GLU,
+              L122.LC_bm2_R_OtherArableLand_Yh_GLU,
+              L123.LC_bm2_R_MgdPast_Yh_GLU,
+              L123.LC_bm2_R_MgdFor_Yh_GLU,
+              L124.LC_bm2_R_Shrub_Yh_GLU_adj,
+              L124.LC_bm2_R_Grass_Yh_GLU_adj,
+              L124.LC_bm2_R_UnMgdPast_Yh_GLU_adj,
+              L124.LC_bm2_R_UnMgdFor_Yh_GLU_adj) %>%
+      # The next two lines are TEMPORARY - because currently we're using temp-data-inject data
+      # When using final data, don't need to reshape them
+      gather(year, value, -GCAM_region_ID, -Land_Type, -GLU) %>%
+      mutate(year = as.integer(substr(year, 2, 5))) %>%
+      bind_rows(L120.LC_bm2_R_UrbanLand_Yh_GLU,
+                L120.LC_bm2_R_Tundra_Yh_GLU,
+                L120.LC_bm2_R_RckIceDsrt_Yh_GLU)  ->
+      L125.LC_bm2_R_LT_Yh_GLU
+
+    L125.LC_bm2_R_LT_Yh_GLU %>%
+      group_by(GCAM_region_ID, GLU, year) %>%               # group the data by GCAM_reigon_id and GLU
+      summarise(value = sum(value)) ->                      # Adding up total land area by region, year, and GLU
+      L125.LC_bm2_R_Yh_GLU
+
+    # prepare the temporary varible for calculating the changing rate
+    L125.LC_bm2_R_Yh_GLU %>%
+      spread(key="year",value=value) %>%                   # convert to wide format
+      ungroup() %>%
+      select(-GCAM_region_ID,-GLU) ->                      # remove the unnecessary columns
+      LC_check1 ->                                         # temporary variable 1
+      LC_check2                                            # temporary variable 2
+
+    LC_check2[,2:ncol(LC_check2)] <- LC_check1[,1:ncol(LC_check2)-1]   # shift the columns of temporary variable 2
+    LC_check <- LC_check1/LC_check2                        # calculate the changing rate
+    LC_check[ is.na( LC_check ) ] <- 1                     # assign na to 1 (no change)
+
+    # stop if the changing rate out of the tolerance boundaries
+    if( any( LC_check < (1 - LANDTOLERANCE) |
+             LC_check > (1 + LANDTOLERANCE) ) )
+      { stop( "ERROR: Interannual fluctuation in global land cover exceeds tolerance threshold" ) }
+
+
+    #Write out the totals, by region and by region x GLU
+    L125.LC_bm2_R_Yh_GLU %>%
+      filter(year==1700) %>%                               # using the starting year only
```

Comment:
Two notes:
1) why are we only looking at the totals in one year, and why is that one year 1700?
2) I think filtering to 1700 explicitly here means we won't be able to shift time. I don't know if that matters or not.